src/SprykerEco/Zed/Braintree/Business/BraintreeFacadeInterface.php

```diff
@@ -233,4 +241,37 @@ public function postSaveHook(QuoteTransfer $quoteTransfer, CheckoutResponseTrans
      * @return \Generated\Shared\Transfer\PaymentMethodsTransfer
      */
     public function filterPaypalExpressPaymentMethod(PaymentMethodsTransfer $paymentMethodsTransfer, QuoteTransfer $quoteTransfer): PaymentMethodsTransfer;
+
+    /**
+     * Specification:
+     * - Checks if Braintree has a nonce.
+     * - QuoteTransfer.Payment and QuoteTransfer.Payment.Braintree must be set.
+     *
+     * @api
+     *
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\CheckoutResponseTransfer $checkoutResponseTransfer
+     *
+     * @return bool
+     */
+    public function isQuotePaymentValid(
+        QuoteTransfer $quoteTransfer,
+        CheckoutResponseTransfer $checkoutResponseTransfer
+    ): bool;
+
+    /**
+     * Specification:
+     * - Executes Braintree sale API request.
+     * - Updates `CheckoutResponseTransfer` and `QuoteTransfer` accordingly to API response.
+     * - If API request is successful - updates order payment method data according to `QuoteTransfer`.
+     * - QuoteTransfer.Payment must be set.
```

Comment:
* Requires `QuoteTransfer.payment` must be set.
* Add a backticks to a transfer object.